# Final Project: Analysis of Heart Disease

# Introduction


Our main objective is to analyse the dataset of attributes related to diabetes and to predict whether that person is diabetic or not. We will be applying machine learning algorithms to try to achieve this goal. 


## Libraries:

In [10]:
import requests
import pandas as pd
import json
import numpy as np

## Data Used
In the data collection stage of the data life cycle, you need to focus on collecting data from websites and databases. 

We have found data from Kaggle at: 
https://www.kaggle.com/datasets/mathchi/diabetes-data-set

This data has all the attributes needed for predicting diabetes which will aid us to reach our final goal. The data is of 21 year old women at the Pima Indian Heritage. The data is organized with the folloing attributes as described by the website: 


Pregnancies: Number of times pregnant

Glucose: Plasma glucose concentration a 2 hours in an oral glucose tolerance test

BloodPressure: Diastolic blood pressure (mm Hg)

SkinThickness: Triceps skin fold thickness (mm)

Insulin: 2-Hour serum insulin (mu U/ml)

BMI: Body mass index (weight in kg/(height in m)^2)

DiabetesPedigreeFunction: Diabetes pedigree function

Age: Age (years)

Outcome: Class variable (0 or 1)




In [9]:
attributes = pd.read_csv("diabetes.csv")
attributes.head(10)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   
5            5      116             74              0        0  25.6   
6            3       78             50             32       88  31.0   
7           10      115              0              0        0  35.3   
8            2      197             70             45      543  30.5   
9            8      125             96              0        0   0.0   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1  
5                     0.201   30        0  
6                     0.248   26        1  
7                     0.134   29        0  
8                     0.158   53        1  
9                     0.232   54        1

As you can see, from the data shown above each individual has an number id and has each attribute filled out.

Outcomes, the most important piece of data is the last column.

In [11]:
attributes.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

As we can see, there are 768 individual data points. There are some interesting points to how much the data points vary for example skin thickness can vary from 23 at the 50th percentile to 99 at the maximum. 

Other things to note is that there are no abnormal conditions seen outside of things expected to see in diabetics, for example the maximum blood preasure is 122 which is within a normal range and isnt considered to be harmful. 

There is an issue when it comes to the lower bounds as there are many 0 values that simply wouldn't be possible to have. It is assumed that these 0 values are simply missing values so we will need to replace them with nan. 

In [12]:
attributes = attributes.replace(0,np.nan)

In [13]:
attributes.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   657.000000  763.000000     733.000000     541.000000  394.000000   
mean      4.494673  121.686763      72.405184      29.153420  155.548223   
std       3.217291   30.535641      12.382158      10.476982  118.775855   
min       1.000000   44.000000      24.000000       7.000000   14.000000   
25%       2.000000   99.000000      64.000000      22.000000   76.250000   
50%       4.000000  117.000000      72.000000      29.000000  125.000000   
75%       7.000000  141.000000      80.000000      36.000000  190.000000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age  Outcome  
count  757.000000                768.000000  768.000000    268.0  
mean    32.457464                  0.471876   33.240885      1.0  
std      6.924988                  0.331329   11.760232      0.0  
min     18.200000                  0.078000   21.000000      1.0  
25%     27.500000                  0.243750   24.000000      1.0  
50%     32.300000                  0.372500   29.000000      1.0  
75%     36.600000                  0.626250   41.000000      1.0  
max     67.100000                  2.420000   81.000000      1.0

Now we have a more accurate representation of the data. 

# Data Processing

## processing the data:
